streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfo.java

```diff
@@ -61,8 +63,8 @@ private AssignmentInfo(final int version,
 
     public AssignmentInfo(final List<TaskId> activeTasks,
                           final Map<TaskId, Set<TopicPartition>> standbyTasks,
-                          final Map<HostInfo, Set<TopicPartition>> hostState) {
-        this(LATEST_SUPPORTED_VERSION, activeTasks, standbyTasks, hostState, 0);
+                          final Map<HostInfo, Set<TaskId>> tasksByHost) {
+        this(LATEST_SUPPORTED_VERSION, activeTasks, standbyTasks, tasksByHost, 0);
```

Comment:
For compatibility, we cannot change any message format of old versions, so I think we need to bump up the version to 5 and only do compression at 5 while no compression at version 4. Think about the case: when a streams application with multiple instances are rolling bounce to be upgraded, maybe some instance are already on version 5 and hence sends the assignment back compressed while some other instance do not recognize this version at all. We need to distinguish the cases when to compress and when to not compress.

@mjsax has done the version probing protocol and he can provide more.

Reply:
Sounds correct

Reply:
If I increment the LATEST_SUPPORTED_VERSION of AssignmentInfo as 5, should LATEST_SUPPORTED_VERSION of SubscriptionInfo be also changed to 5.

Reply:
I think the versions of subscriptionInfor / assignmentInfo are coupled, so if we bump up on one side we should do the same on the other.

Reply:
Agreed. (I don't think they are technically coupled, but it would simplify our live significantly if we we keep them in sync -- I was considering to unify both in one place at some point, but thought it's also ok if we just maintain both correctly.)